api/db/users.js

```diff
@@ -101,21 +101,35 @@ const getAllUsers = async ({
 
 const getUserByID = async (
   id,
+  checkOkta,
   {
     clean = true,
     client = oktaClient,
     populate = populateUser,
-    additionalValues
+    additionalValues,
+    db
   } = {}
 ) => {
-  const oktaUser = await client.getUser(id);
 
+  let oktaUser
```

Comment:
I think this can be a `const` since it's never reassigned to another type other than `object`. I'm not sure what all client.getUser can return though.

Reply:
If I am honest, this is where I find the difference between const and let highly confusing.  In my mind const should be used when something will not ever change.  const ACTIVITY = "ACTIVITY", or const people = getPeople() make sense, because once I get that thing, I don't intend to change it and I signal that with a const initializer.  When I see `const people = getPeople()
people.push(newPerson)
` 
I feel like const is misplaced and should be let so that the next dev knows this will in fact change so they need to follow it through the code to know what it will be.

Reply:
I think this is a valid use of `let` because you are setting it differently depending on the status of the boolean and you are doing something else in addition, otherwise you could set it using a ternary operation. You could avoid this by making this section a separate function and returning oktaUser from it.